doc/smart-switch/gnmi-feedback/smart-switch-gnmi-feedback-design.md

```diff
@@ -0,0 +1,158 @@
+
+## Requirement
+
+- Support for multiple operations in a single request
+- Non-child objects must always include the version ID field
+- The GNMI server supports set, remove, get, and subscribe operations
+- The GNMI client can proactively subscribe to or get objects/tables via object key or table name
+
+| Child object      |
+| ----------------- |
+| DASH_ACL_RULE     |
+| DASH_ROUTE        |
+| DASH_ROUTE_RULE   |
+| DASH_VNET_MAPPING |
+
+
+## Sequence flow
+
+### Set/Remove
+
+```mermaid
+sequenceDiagram
+    participant GC as GNMI Client
+    box NPU SONiC
+        participant GS as GNMI Server
+        participant AD as APP DB
+        participant ASD as APP STATE DB
+    end
+    Note over AD, ASD: DPUx Redis
+    box DPU SONiC
+        participant ZMQ as ZMQ
+        participant SW as SWSS
+        participant SY as SYNCD
+    end
+
+    GC ->> GS: Update commands
+    loop Each commands:
+        GS --) AD: Insert object
+        GS ->> ZMQ: Update command
+    end
+    GS -->> GC: 
+    ZMQ ->> SW: Consume commands
+    SW ->> SY: SAI call
+    SY -->> SW: 
+    alt Set command
+        SW ->> ASD: Update command result
+        Note over ZMQ: Include result code and version ID (If it's non-child object)
+    else Del command
+        SW ->> ASD: Remove objects
+    end
+
+```
+
+### Message
+
+### Get
+
+```mermaid
+sequenceDiagram
+    participant GC as GNMI Client
+    box NPU SONiC
+        participant GS as GNMI Server
+        participant ASD as APP STATE DB
+    end
+    Note over ASD: DPUx Redis
+
+    GC ->> GS: Query object or table result
+    GS ->> ASD: Get object via key or get objects via table name
+    ASD -->> GS: Query results
+    GS -->> GC: Query result
+
+```
+
+### Subscribe
+
+```mermaid
+sequenceDiagram
+    participant GC as GNMI Client
+    box NPU SONiC
+        participant GS as GNMI Server
+        participant ASD as APP STATE DB
+    end
+    Note over ASD: DPUx Redis
+    participant DPU as DPU
+
+    GC ->> GS: Subscribe object or table
+    GS ->> GS: Register subscribe events
+    GS ->> ASD: Get object via key or get objects via table name
+    Note over ASD: Cache result before subscribe
+    ASD -->> GS: Query results
+    GS -->> GC: Query results
+    loop Each commands
+        alt Set command
+            DPU ->> ASD: Update command result
+        else Del command
+            DPU ->> ASD: Remove objects
+        end
+    end
+    ASD -->> GS: Proactively push results
+    GS -->> GC: Proactively push results
+
+```
+
+## Table
+
+### APPL STATE DB
+
+DASH state table in APPL STATE DB, table name is the same as DASH table, key is the same, and fields are result and version ID if it's not a child object.
```

Comment:
Add an example for DASH_ROUTE_GROUP_TABLE?

Reply:
Here is just an example, more tables will be on the dash hld document.